src/librustc_mir/transform/check_consts/validation.rs

```diff
@@ -725,3 +691,37 @@ fn check_return_ty_is_sync(tcx: TyCtxt<'tcx>, body: &Body<'tcx>, hir_id: HirId)
         }
     });
 }
+
+fn place_as_reborrow(
+    tcx: TyCtxt<'tcx>,
+    body: &Body<'tcx>,
+    place: &'a Place<'tcx>,
+) -> Option<&'a [PlaceElem<'tcx>]> {
+    place
+        .projection
+        .split_last()
+        .and_then(|(outermost, inner)| {
+            if outermost != &ProjectionElem::Deref {
+                return None;
+            }
+
+            // A borrow of a `static` also looks like `&(*_1)` in the MIR, but `_1` is a `const`
+            // that points to the allocation for the static. Don't treat these as reborrows.
+            if let PlaceBase::Local(local) = place.base {
+                let decl = &body.local_decls[local];
+                if let LocalInfo::StaticRef { .. } = decl.local_info {
+                    return None;
+                }
+            }
```

Comment:
@matthewjasper isn't this supposed to use a helper method?

Reply:
Yes, there's `body.local_decls[local].is_ref_to_static`

Reply:
@matthewjasper Resolved in ccb4eed.